### ISSUE TITLE
differentiable graph creation

### ISSUE THREAD

**Issue by jakubMitura14** [axelparmentier/InferOpt.jl#23]
Hello I have a task to create a graph from the output of the convolutional neural network that then will be used in graph neural network.
I have a cost function that can tell wheather graph was well created.
My main problem is that graph by its nature is discrete hence constructing graph creation algorithm in a way that will enable back propagation is problematic (as you see in my architecture gradients need to pass from graph neural networks to CNN) 

Can it be potentially possible to use your package for differentiable graph creation from 3D array?

**Comment by rejuvyesh** [comment]
Depending on how the graph edge is represented, a `gumble softmax` is the usual thing to do in that case to differentiate through your graph edge predictions.

**Comment by jakubMitura14** [comment]
thank You !! I will investigate this path!
